src/wallet/wallet.cpp

```diff
@@ -1874,106 +1888,109 @@ CAmount CWalletTx::GetDebit(const isminefilter& filter) const
     if(filter & ISMINE_SPENDABLE)
     {
         if (fDebitCached)
-            debit += nDebitCached[ColorIdentifier()];
+            debit += nDebitCached[colorId];
         else
         {
-            nDebitCached[ColorIdentifier()] = pwallet->GetDebit(*tx, ISMINE_SPENDABLE);
+            nDebitCached = pwallet->GetDebit(*tx, ISMINE_SPENDABLE);
             fDebitCached = true;
-            debit += nDebitCached[ColorIdentifier()];
+            debit += nDebitCached[colorId];
         }
     }
     if(filter & ISMINE_WATCH_ONLY)
     {
         if(fWatchDebitCached)
-            debit += nWatchDebitCached[ColorIdentifier()];
+            debit += nWatchDebitCached[colorId];
         else
         {
-            nWatchDebitCached[ColorIdentifier()] = pwallet->GetDebit(*tx, ISMINE_WATCH_ONLY);
+            nWatchDebitCached = pwallet->GetDebit(*tx, ISMINE_WATCH_ONLY);
             fWatchDebitCached = true;
-            debit += nWatchDebitCached[ColorIdentifier()];
+            debit += nWatchDebitCached[colorId];
         }
     }
     return debit;
 }
 
-CAmount CWalletTx::GetCredit(const isminefilter& filter) const
+CAmount CWalletTx::GetCredit(const isminefilter& filter, ColorIdentifier& colorId) const
 {
-    CAmount credit = 0;
+    TxColoredCoinBalancesMap credit;
+    credit[colorId] = 0;
     if (filter & ISMINE_SPENDABLE)
     {
         // GetBalance can assume transactions in mapWallet won't change
         if (fCreditCached)
-            credit += nCreditCached[ColorIdentifier()];
+            credit[colorId] += nCreditCached[colorId];
         else
         {
-            nCreditCached[ColorIdentifier()] = pwallet->GetCredit(*tx, ISMINE_SPENDABLE);
+            nCreditCached = pwallet->GetCredit(*tx, ISMINE_SPENDABLE);
             fCreditCached = true;
-            credit += nCreditCached[ColorIdentifier()];
+            credit[colorId] += nCreditCached[colorId];
         }
     }
     if (filter & ISMINE_WATCH_ONLY)
     {
         if (fWatchCreditCached)
-            credit += nWatchCreditCached[ColorIdentifier()];
+            credit[colorId] += nWatchCreditCached[colorId];
         else
         {
-            nWatchCreditCached[ColorIdentifier()] = pwallet->GetCredit(*tx, ISMINE_WATCH_ONLY);
+            nWatchCreditCached = pwallet->GetCredit(*tx, ISMINE_WATCH_ONLY);
             fWatchCreditCached = true;
-            credit += nWatchCreditCached[ColorIdentifier()];
+            credit[colorId] += nWatchCreditCached[colorId];
         }
     }
-    return credit;
+    return credit[colorId];
 }
 
-CAmount CWalletTx::GetAvailableCredit(bool fUseCache, const isminefilter& filter) const
+TxColoredCoinBalancesMap CWalletTx::GetAvailableCredit(bool fUseCache, const isminefilter& filter) const
 {
+    TxColoredCoinBalancesMap nCredit;
+    nCredit[ColorIdentifier()] = 0;
```

Comment:
Removing this initialization leads GUI test failure. 
This would be the same issue I fixed in 0cb6b68eef68f63dc7be2096c4a8939387029c9b.  

I will fix this kind of issue.

Reply:
fixed in 5aaa9d88e4ba65c0b6455b8a0c6d787ca688ee45